mindsdb/integrations/handlers/chromadb_handler/chromadb_handler.py

```diff
@@ -122,144 +93,88 @@ def check_connection(self):
-                    ),  # filters on metadata, for now only column name
-                    query=where.get("search_query"),
-                    k=query.limit.value if query.limit else 5,
+                collection_data = self._collection.query(
+                    query_embeddings=search_embedding,
+                    n_results=query.limit.value if query.limit else 5,
+                    include=["embeddings", "distances"],
                 )
-                result = documents_to_df(collection_data)
+
+                embeddings_arr = np.array(collection_data["embeddings"][0])
+                distance_arr = np.array(collection_data["distances"][0])
+
+                indices = np.where(distance_arr > 0.5)[0]
+                result["embeddings"] = embeddings_arr[indices].tolist()
 
             else:
-                # if no 'search_query' provided, run a regular query and filter on metadata if provided
-                collection_data = self._collection.get(
-                    where=where.get("meta_data_filter")
-                )
-                result = pd.DataFrame(collection_data)[
-                    ["ids", "documents", "metadatas"]
-                ]
 
-        if query.limit:
-            # if there is a limit clause, limit the result.
-            result = result.head(query.limit.value)
+                return Response(
+                    resp_type=RESPONSE_TYPE.ERROR,
+                    error_message="SELECT only supports COUNT(*), SELECT * or WHERE with 'search_embeddings' parameter",
+                )
 
         return Response(resp_type=RESPONSE_TYPE.TABLE, data_frame=result)
 
-    def run_create_table(self, query: ASTNode) -> Response:
+    def create_or_get_collection(self, query: ASTNode) -> Response:
         """
         Run a create table query on the ChromaDB database.
         """
         collection_name = query.name.parts[-1]
-        self._client.create_collection(collection_name)
+        self._client.get_or_create_collection(collection_name)
 
         return Response(resp_type=RESPONSE_TYPE.OK)
 
-    def run_insert(self, query):
+    def add_embeddings(self, query):
         """
         Run an insert query on the ChromaDB database.
         """
         collection_name = query.table.parts[-1]
-        columns = [column.name for column in query.columns]
-        df = pd.DataFrame(data=query.values, columns=columns)
-        documents = split_documents(df, columns)
-
-        # converts list of Documents to embedding vectors and stores them in a ChromaDB collection
-        Chroma.from_documents(
-            documents=documents,
-            embedding=self._embedding_function,
-            persist_directory=self._persist_directory,
-            client_settings=Settings(**self._client_config),
-            collection_name=collection_name,
-        )
+
+        embeddings = [ast.literal_eval(embedding[0]) for embedding in query.values]
+        ids = [str(uuid.uuid1()) for _ in embeddings]
```

Comment:
Can we check if users have already provided `id` columns to us, so we may don't need to auto generate for them?

Can we also support users providing `content` (optional), `metadata` (optional) columns?